htmlelements-java/src/main/java/ru/yandex/qatools/htmlelements/loader/decorator/HtmlElementLocatorFactory.java

```diff
@@ -40,6 +44,31 @@ public ElementLocator createLocator(Field field) {
      * @param clazz Class for which locator will be created.
      */
     public ElementLocator createLocator(Class clazz) {
-        return new AjaxElementLocator(searchContext, new HtmlElementClassAnnotationsHandler(clazz));
+        return new AjaxElementLocator(searchContext, getTimeOut(clazz), new HtmlElementClassAnnotationsHandler(clazz));
+    }
+
+    public int getTimeOut(Field field) {
+        if (field.isAnnotationPresent(Timeout.class)) {
+            return field.getDeclaredAnnotation(Timeout.class).value();
+        }
+        if (field.getGenericType() instanceof Class) {
+            return getTimeOut((Class) field.getGenericType());
+        }
+        return getTimeOut((Class) ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0]);
+    }
+
+    public int getTimeOut(Class clazz) {
+        try {
+            Method method = Timeout.class.getMethod("value");
+            do {
+                if (clazz.isAnnotationPresent(Timeout.class)) {
+                    return (Integer) method.invoke(clazz.getDeclaredAnnotation(Timeout.class));
+                }
+                clazz = clazz.getSuperclass();
+            } while (clazz != Object.class && clazz != null);
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+        }
+
+        return 1;
```

Comment:
we should return 5 here to keep backward compatibility with previous implementation